m2cgen/exporters.py

```diff
@@ -251,6 +251,29 @@ def export_to_php(model, indent=4):
     return _export(model, interpreter)
 
 
+def export_to_dart(model, class_name="Model", indent=4):
```

Comment:
Please check the following comment: https://github.com/BayesWitnesses/m2cgen/pull/97#pullrequestreview-287153120. Feel free to ask for any help if you find something unclear in `e2e` tests.

Reply:
Of course! I knew I missed something!

Reply:
One question I have is do these tests get run manually by a dev as a check or in the build pipeline of the package? My latest commit resulted in a successful build and it did not include the e2e tests. So, as long as the e2e DartExecutor passes on my system, that should suffice?

Reply:
Once `DartExecutor` (with some corresponding changes) added to the package, new commits will be run at Travis with Dart e2e tests.

![image](https://user-images.githubusercontent.com/25141164/74621139-c43e4100-514c-11ea-84c9-1de802492354.png)

Reply:
I see that the code is running a command to test the script using the cli tool of that particular language. Do I need to provide information somewhere as to how to set up that cli tool in Travis? I'm afk right now so I don't have a chance to go through that directory to see if that exists.

Reply:
@izeigerman or @StrikerRUS 
It appears the Dart language will throw an error if there is a copious amount of nesting in the code. This makes 3 tests fail, most likely because they are more complex models.

```
Error: The file has too many nested expressions or statements.
E           Try simplifying the code.
E               return ((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((21.24023452121809) + ((exp((var0) * (((((((((((((pow((15.0234) - (input[0]), 2)) + (pow((0.0) - (input[1]), 2))) + (pow((18.1) - (input[2]), 2))) + (pow((0.0) - (input[3]), 2))) + (pow((0.614) - (input[4]), 2))) + (pow((5.304) - (input[5]), 2))) + (pow((97.3) - (input[6]), 2))) + (pow((2.1007)...
```
<img width="418" alt="Screen Shot 2020-02-18 at 8 46 44 PM" src="https://user-images.githubusercontent.com/12801051/74793966-e8ef0180-528f-11ea-8d4c-663a73661d9c.png">

Any thoughts? Obviously, there is no reason to change the interpreter, but is there a way to silence these tests and if so, is that permitted? This is a constraint on the language not m2cgen.

Reply:
@MattConflitti Thank you very much for setting up e2e tests for Dart language! You did it without any guidance from us, impressive! You rock!

Speaking about 3 failing tests, it seems that you've encountered the same problem as me for R language some time ago. Please refer to https://github.com/BayesWitnesses/m2cgen/pull/139#issuecomment-568942961. As a workaround, Dart interpreter should inherit `mixins.BinExpressionDepthTrackingMixin` class and you should play around with the default value for `bin_depth_threshold` which will allow to generate all types of models with reasonable complexity. Also, refer to https://github.com/BayesWitnesses/m2cgen/pull/139#discussion_r361502593.

Reply:
Great! I will make that change and see if I can get it to pass! I appreciate the guidance!

Reply:
It appears to have worked. Should I use the subroutine mixin as well? If I understand it correctly, it is essentially building proxy functions to call so that there is less nesting? Is that necessary if the bin threshold works up to 465 for Dart? Thanks!

Reply:
> Should I use the subroutine mixin as well?

No as long as there are no restrictions on a size of the individual function/method in DART.

> Is that necessary if the bin threshold works up to 465 for Dart? 

Not really. `BinExpressionDepthTrackingMixin ` and the Subroutine mixin address somewhat different problems. The former workarounds the constraints on the depth of binary expressions while the later helps overcome function/method size constraints.